spanner/client_test.go

```diff
@@ -2143,12 +2344,10 @@ func TestFailedUpdate_ShouldRollback(t *testing.T) {
 	if got, want := status.Convert(err).Code(), codes.InvalidArgument; got != want {
 		t.Fatalf("Error mismatch\nGot: %v\nWant: %v", got, want)
 	}
-	// The failed update should trigger a rollback.
+	// No rollback request will be initiated because the client does not receive any transactionId.
```

Comment:
This is not the correct behavior (although in this specific case it does not matter). If the first statement in a transaction fails, then we should restart the transaction and use an explicit BeginTransaction RPC. Otherwise, if this transaction would contain more statements, the following statement in the transaction would include the BeginTransaction option. That again would mean that the first statement seems to have been executed on the transaction, while it actually hasn't.

See also the other comments.

Reply:
Ack, fixed